### PR TITLE
enable user-configurable Azure CNI URL

### PR DESCRIPTION
**Pull request by jackfrancis**
**What this PR does / why we need it**: Enables user-configurable Azure CNI URL. Linux implementation is code-sane, but doesn't actually do anything, as that'll require more work integrating w/ VHD. Because Windows is the desired initial use-case, I'm advocating we take this as-is (once we prove Windows functionality) and leave remaining Linux implementation to be done when desired.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**If applicable**:
- [ ] documentation
- [ ] unit tests
- [ ] tested backward compatibility (ie. deploy with previous version, upgrade with this branch)

**Release note**:

```
enable user-configurable Azure CNI URL
```

### CODE REVIEW

**Comment by jsturtevant** [comment]
fixes: https://github.com/Azure/acs-engine/issues/4007

File: pkg/api/types.go [hunk 1]
```diff
@@ -360,6 +360,8 @@ type KubernetesConfig struct {
 	LoadBalancerSku                  string            `json:"loadBalancerSku,omitempty"`
 	ExcludeMasterFromStandardLB      *bool             `json:"excludeMasterFromStandardLB,omitempty"`
 	AzureCNIVersion                  string            `json:"azureCNIVersion,omitempty"`
+	AzureCNIURLLinux                 string            `json:"azureCNIURLLinux,omitempty"`
```
Comment (tariq1890) [inline pkg/api/types.go]: Can we have a struct (AzureCNIConfig perhaps? )that wraps AzureCNIVersion, AzureCIURLLinux, AzureCNIURLWindows, and AzureCNIPluginURL ? This way we make all the four user configurable and struct can be reused in KubernetesConfig as well as KubernetesSpecConfig structs.

KubernetesConfig is getting really bloated right now :(

**Comment by jsturtevant** [comment]
reference back to epic #3851 for tracking

**Pull request merged**